src/flags/FlagStore.ts

```diff
@@ -0,0 +1,89 @@
+import { reportError } from '../drivers/SentryConnector';
+import { Db, ObjectId } from 'mongodb';
+import { MongoDriver } from '../drivers/MongoDriver';
+import { FlagDataStore } from './interfaces/FlagDataStore';
+import { Flag } from '../types/Flag';
+
+enum Collections {
+    FLAGS = 'flags',
+    RATINGS = 'ratings',
+}
+export class FlagStore implements FlagDataStore {
+    private static instance: FlagDataStore;
+    private db: Db;
+
+    private constructor() {
+        this.db = MongoDriver.getConnection();
+    }
+
+    static getInstance(): FlagDataStore {
+        if (!FlagStore.instance) {
```

Comment:
Should we use `this` here instead of `FlagStore`? I see in a later section of these changes that `this` is used in a static method over `FlagStore`.
```suggestion
      if (!this.instance) {
```